artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/HierarchicalObjectRepository.java

```diff
@@ -94,6 +99,11 @@
     */
    private final ArrayList<HierarchicalRepositoryChangeListener> listeners = new ArrayList<>();
 
+   public HierarchicalObjectRepository(final WildcardConfiguration wildcardConfiguration) {
```

Comment:
I'd rather see a default constructor which calls this one so all the places in the code-base which use the default WildcardConfiguration don't have to be changed to pass in `new WildcardConfiguration()` or `null`.

Reply:
Fair enough I was divided at the time between the two approaches, will change

Reply:
have updated PR with this.